src/resolver.c

```diff
@@ -48,14 +48,22 @@
  * Forward declarations.
  ******************************************************************************/
 
+struct resolver_ctx {
+    xmpp_ctx_t *ctx;
+    const char *domain;
+    resolver_srv_rr_t **srv_rr_list;
+    void (*callback)(xmpp_sock_t *xsock, const char *domain);
+    xmpp_sock_t *xsock;
+};
+
 #ifdef HAVE_CARES
+ares_channel ares_chan = NULL;
```

Comment:
c-ares suggests to only have one ares_channel and use it for everything in the whole program. It is locked by ares and thus threadsafe and so having a global here should be safe.